### DISCUSSION TITLE
The interactions in a 2D interactive diagram is not displayed

### DISCUSSION THREAD

**Discussion by sribone** [chemosim-lab/ProLIF#227] [Q&A]
Dear all, 
I'm running prolif to performed interactions fingerprints. When I run the fp.plot_barcode() works just fine showing me the barcode plot, but when I run the fp.plot_lignetwork it doesn't work... I mean the analysis is done because I can see the percentage bar progressing but the 2D interaction diagram is never showed. This is my script for running both graphics:

# load topology and trajectory
    u = mda.Universe("com.prmtop", "prod-strip.nc")
    # create selections for the ligand and protein
    ligand_selection = u.select_atoms("resname MOL")

    
    ligand_mol = plf.Molecule.from_mda(ligand_selection)
    protein_selection = u.select_atoms("protein")
    
    # # use default interactions
    list_of_interactions = ['Anionic','CationPi','Cationic','HBAcceptor','HBDonor','Hydrophobic','MetalAcceptor','MetalDonor','PiCation','PiStacking','VdWContact','XBAcceptor','XBDonor']
    fp = plf.Fingerprint(list_of_interactions)
    # run on a slice of the trajectory frames: from begining to end with a step of 10
    fp.run(u.trajectory[::framestep], ligand_selection, protein_selection)
    fp.to_pickle(fingeprint_file)
    df = fp.to_dataframe(drop_empty=False)
    fp.plot_barcode()
    fp.plot_lignetwork(ligand_mol)
 
Do you know what can be producing this problem?
Thank you very much!!

**Comment by cbouy** [comment]
Hi @sribone 

The LigNetwork plot only works in Jupyter notebooks, not in plain python scripts unfortunately.
However you can save it to an HTML file and then display it in your web browser, just replace `fp.plot_lignetwork(ligand_mol)` with:
```python
from prolif.plotting.network import LigNetwork

ligplot = LigNetwork.from_fingerprint(fp, ligand_mol)
ligplot.save("ligplot.html")
```

I'll close this issue but if this doesn't solve your problem feel free to reopen!
